packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -967,8 +967,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
          * we don't have an op (on demand summaries for instance). In those cases, we will use the timestamp
          * of this client's connection - https://github.com/microsoft/FluidFramework/issues/8375.
          */
-        const getCurrentTimestamp = () => {
-            return this.deltaManager.lastMessage?.timestamp ?? Date.now();
+         const getCurrentTimestamp = () => {
+            const audience = this.getAudience();
+            const client = this.clientId !== undefined ? audience.getMember(this.clientId) : undefined;
```

Comment:
Ideally we should get the client from audience when we move to "connected" state and store the timestamp in a member called `connectedTimestamp` or something similar.
However, as Vlad pointed out the client may not be in the audience when we move to connected state - https://github.com/microsoft/FluidFramework/issues/8375#issuecomment-996195788.

We can change this once https://github.com/microsoft/FluidFramework/issues/7275 is fixed. Can you add a comment to 7275 to update this code once it is fixed?

Reply:
Or perhaps create a new issue to change this once #7275 is fixed.